app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,101 @@
+class OrdersController < ApplicationController
+  before_action :account_views, only: [:edit]
+
+  def index
+    @orders = Order.all
+  end
+
+  def new
+    @user = User.find_by(id: session[:user_id])
+
+    if @user.nil?
+      if User.find_by(id: 1)
+        @user = User.find_by(id: 1)
+      else
+        @user = User.new
+        @user.name = "Guest"
+        @user.id = 1
+      end
+    end
+
+    session[:user_id] = @user.id
+
+    @order = Order.new(user_id: @user.id)
+  end
+
+  def show
+    @order = Order.find_by(id: params[:id])
+    @user = User.find_by(uid: session['uid'])
+
+    if @order.user_id != @user.id
+      flash[:status] = :failure
+      flash[:result_text] = "This is not your order. Merchants please access your sold oders through your orders page"
+      flash[:messages] = @order.errors.messages
+      redirect_to :root
+    end
+
+    if @user
+      @order_items = OrderItem.where(order_id: @order.id)
+    else
+      product_ids = @user.products.map{ |i| i.id }
```

Comment:
When would the code ever get into this branch?

If `@user` is falsey (is `nil`) and gets into this `else` block, why would we be able to call `product_ids = @user.products.map{ |i| i.id }`?